android/app/src/main/java/co/ello/ElloApp/MainActivity.java

```diff
@@ -338,13 +338,13 @@ private void setupRegisterDeviceReceiver() {
             @Override
             public void onReceive(Context context, Intent intent) {
                 String reg_id = intent.getExtras().getString("GCM_REG_ID");
-                String registerFunctionCall =
+                if(reg_id != null) {
+                    String registerFunctionCall =
                         "javascript:registerAndroidNotifications(\"" +
                                 reg_id + "\", \"" +
                                 packageName() + "\", \"" +
                                 versionName() + "\", \"" +
                                 versionCode() + "\")";
-                if(reg_id != null) {
```

Comment:
@steam if I'm reading the crash and [API docs](https://developer.android.com/reference/android/content/Intent.html#getExtras()) correctly, I think the crash is actually from `getExtras()` returning `null` instead of a `Bundle`. I think you may need to guard against that too.